utils/upload_ansible_roles_to_galaxy.py

```diff
@@ -82,132 +83,178 @@ def clone_and_init_repository(parent_dir, repo):
-    local_readme_content = local_readme_content.replace(
-        "@MIN_ANSIBLE_VERSION@", ssg.ansible.min_ansible_version)
-    local_readme_content = local_readme_content.replace("@ROLE_NAME@",
-                                                        repository.name)
-
-    remote_readme_file = repository.get_file_contents("/README.md")
-
-    if local_readme_content != remote_readme_file.decoded_content:
-        print("Updating README.md in %s" % repository.name)
-
-        repository.update_file(
-            "/README.md",
-            "Updates README.md",
-            local_readme_content,
-            remote_readme_file.sha,
-            author=InputGitAuthor(
-                GIT_COMMIT_AUTHOR_NAME, GIT_COMMIT_AUTHOR_EMAIL)
-        )
+    def _update_meta_content_if_needed(self):
+        with open(META_TEMPLATE_PATH, 'r') as f:
+            meta_template = f.read()
+
+        local_meta_content = meta_template.replace("@DESCRIPTION@", self.title)
+        local_meta_content = local_meta_content.replace(
+            "@MIN_ANSIBLE_VERSION@", ssg.ansible.min_ansible_version)
+        remote_meta_file = self.remote_repo.get_file_contents("/meta/main.yml")
+
+        if local_meta_content != remote_meta_file.decoded_content:
+            print("Updating meta/main.yml in %s" % self.remote_repo.name)
+            self.remote_repo.update_file(
+                "/meta/main.yml",
+                "Updates meta/main.yml",
+                local_meta_content,
+                remote_meta_file.sha,
+                author=InputGitAuthor(
+                    GIT_COMMIT_AUTHOR_NAME, GIT_COMMIT_AUTHOR_EMAIL)
+            )
 
-    with open(META_TEMPLATE_PATH, 'r') as f:
-        meta_template = f.read()
-
-    local_meta_content = meta_template.replace("@DESCRIPTION@", title)
-    local_meta_content = local_meta_content.replace(
-        "@MIN_ANSIBLE_VERSION@", ssg.ansible.min_ansible_version)
-    remote_meta_file = repository.get_file_contents("/meta/main.yml")
-
-    if local_meta_content != remote_meta_file.decoded_content:
-        print("Updating meta/main.yml in %s" % repository.name)
-        repository.update_file(
-            "/meta/main.yml",
-            "Updates meta/main.yml",
-            local_meta_content,
-            remote_meta_file.sha,
-            author=InputGitAuthor(
-                GIT_COMMIT_AUTHOR_NAME, GIT_COMMIT_AUTHOR_EMAIL)
+    def update_repository(self):
+        print("Processing %s..." % self.remote_repo.name)
+
+        self.gather_data()
+        self._reformat_local_content()
+        self.title = re.search(
+            r'Profile Title:\s+(.+)$', self.description, re.MULTILINE).group(1)
+        # Why so?
+        self.description = self.description.replace('\n', '  \n')
+
+        self._update_tasks_content_if_needed()
+        self._update_vars_content_if_needed()
+        self._update_readme_content_if_needed()
+        self._update_meta_content_if_needed()
```

Comment:
Updating the meta content should only be done when the repository is first initialized with the content. Reason being is the meta tags are completely wrong and will deviate by profile to match accurately what the profile is and does.